velox/functions/prestosql/aggregates/tests/MaxSizeForStatsTest.cpp

```diff
@@ -0,0 +1,261 @@
+       "\"$internal$max_data_size_for_stats\"(c6)",
+       "\"$internal$max_data_size_for_stats\"(c7)",
+       "\"$internal$max_data_size_for_stats\"(c8)",
+       "\"$internal$max_data_size_for_stats\"(c9)"},
+      "SELECT * FROM (VALUES (1,1,2,4,8,4,8,1,4,16),(2,1,2,4,8,4,8,1,4,16))");
+
+  // Without grouping keys.
+  testAggregations(
+      vectors,
+      {},
+      {"\"$internal$max_data_size_for_stats\"(c1)",
+       "\"$internal$max_data_size_for_stats\"(c2)",
+       "\"$internal$max_data_size_for_stats\"(c3)",
+       "\"$internal$max_data_size_for_stats\"(c4)",
+       "\"$internal$max_data_size_for_stats\"(c5)",
+       "\"$internal$max_data_size_for_stats\"(c6)",
+       "\"$internal$max_data_size_for_stats\"(c7)",
+       "\"$internal$max_data_size_for_stats\"(c8)",
+       "\"$internal$max_data_size_for_stats\"(c9)"},
+      "SELECT * FROM (VALUES (1,2,4,8,4,8,1,4,16))");
+}
+
+// Input:
+// c0(array(bigint))
+// [1,2,3,4,5]
+// []
+// [1,2,3]
+// Query:
+// select aggr(c0)
+// Output
+// 44
+TEST_F(MaxSizeForStatsTest, arrayGlobalAggregate) {
+  auto vectors = {makeRowVector({makeArrayVector<int64_t>({
+      {1, 2, 3, 4, 5},
+      {},
+      {1, 2, 3},
+  })})};
+  testAggregations(
+      vectors, {}, {"\"$internal$max_data_size_for_stats\"(c0)"}, "SELECT 44");
+}
+
+// Input:
+// c0(map(tinyint,integer))
+// map(array(1,1,1,1,1),array(1,1,1,1,1))
+// map()
+// map(array(1,1,1), array(1,1,1))
+// Query:
+// select aggr(c0)
+// Output
+// 29
+TEST_F(MaxSizeForStatsTest, mapGlobalAggregate) {
+  auto vectors = {makeRowVector({makeMapVector<int8_t, int32_t>(
+      {{{1, 1}, {1, 1}, {1, 1}, {1, 1}, {1, 1}},
+       {},
+       {{1, 1}, {1, 1}, {1, 1}}})})};
+  testAggregations(
+      vectors, {}, {"\"$internal$max_data_size_for_stats\"(c0)"}, "SELECT 29");
+}
+
+// Input:
+// c0( row(array(bigint), map(tinyint,integer))
+// row(aray[1,2,3,4,5], map(array(1,1,1,1,1),array(1,1,1,1,1)))
+// row(array[], map())
+// row(array[1,2,3], map(array(1,1,1), array(1,1,1)))
+// Query:
+// select aggr(c0)
+// Output
+// 77
+TEST_F(MaxSizeForStatsTest, rowGlobalAggregate) {
+  auto vectors = {makeRowVector({makeRowVector(
+      {makeArrayVector<int64_t>({
+           {1, 2, 3, 4, 5},
+           {},
+           {1, 2, 3},
+       }),
+       makeMapVector<int8_t, int32_t>(
+           {{{1, 1}, {1, 1}, {1, 1}, {1, 1}, {1, 1}},
+            {},
+            {{1, 1}, {1, 1}, {1, 1}}})})})};
+  testAggregations(
+      vectors, {}, {"\"$internal$max_data_size_for_stats\"(c0)"}, "SELECT 77");
+}
+
+// Input:
+// c0(varbinary)
+// "buf"
+// ""
+// "hello"
+// query:
+// select aggr(c0)
+// output:
+// 9
+TEST_F(MaxSizeForStatsTest, varbinaryGlobalAggregate) {
+  VectorPtr varbinary_vector = BaseVector::create(VARBINARY(), 3, pool());
+  auto flat_vector = varbinary_vector->asFlatVector<StringView>();
+  StringView sv1{"buf"};
+  StringView sv2{""};
+  StringView sv3{"henlo"};
+  flat_vector->set(0, sv1);
```

Comment:
use "buf"_sv, ""_sv, "hello"_sv

It would be nice to also test with non-inline strings (strings of length > 12).